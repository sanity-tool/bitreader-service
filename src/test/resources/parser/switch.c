static int I;

static void testSwitchWithDefault() {
    switch (I) {
        case 1:
            foo1();
            break;
        case 2:
            foo2();
            break;
        default:
            fooDefault();
    }
    fooAfter();
}

static void testSwitchNoDefault() {
    switch (I) {
        case 1:
            foo1();
            break;
        case 2:
            foo2();
            break;
    }
    fooAfter();
}

static void testSwitchFallthrough() {
    switch (I) {
        case 1:
            foo1();
        case 2:
            foo2();
    }
    fooAfter();
}