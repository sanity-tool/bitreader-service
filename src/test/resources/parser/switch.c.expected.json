{
  "functions" : [ {
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "operands" : [ {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMSwitch",
        "operands" : [ {
          "kind" : "LLVMInstructionValueKind"
        }, {
          "kind" : "LLVMBasicBlockValueKind"
        }, {
          "kind" : "LLVMConstantIntValueKind"
        }, {
          "kind" : "LLVMBasicBlockValueKind"
        }, {
          "kind" : "LLVMConstantIntValueKind"
        }, {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMBr",
        "operands" : [ {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMBr",
        "operands" : [ {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMBr",
        "operands" : [ {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMRet"
      } ]
    } ],
    "entryBlockIndex" : 0
  }, {
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "operands" : [ {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMSwitch",
        "operands" : [ {
          "kind" : "LLVMInstructionValueKind"
        }, {
          "kind" : "LLVMBasicBlockValueKind"
        }, {
          "kind" : "LLVMConstantIntValueKind"
        }, {
          "kind" : "LLVMBasicBlockValueKind"
        }, {
          "kind" : "LLVMConstantIntValueKind"
        }, {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMBr",
        "operands" : [ {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMBr",
        "operands" : [ {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMRet"
      } ]
    } ],
    "entryBlockIndex" : 0
  }, {
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "operands" : [ {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMSwitch",
        "operands" : [ {
          "kind" : "LLVMInstructionValueKind"
        }, {
          "kind" : "LLVMBasicBlockValueKind"
        }, {
          "kind" : "LLVMConstantIntValueKind"
        }, {
          "kind" : "LLVMBasicBlockValueKind"
        }, {
          "kind" : "LLVMConstantIntValueKind"
        }, {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMBr",
        "operands" : [ {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMBr",
        "operands" : [ {
          "kind" : "LLVMBasicBlockValueKind"
        } ]
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        } ]
      }, {
        "kind" : "LLVMRet"
      } ]
    } ],
    "entryBlockIndex" : 0
  } ]
}